connector/src/main/java/org.apache.flink/connector/nebula/sink/NebulaBatchOutputFormat.java

```diff
@@ -106,8 +92,12 @@ public void open(int i, int i1) throws IOException {
                     "nebula-write-output-format"));
             this.scheduledFuture = this.scheduler.scheduleWithFixedDelay(() -> {
                 synchronized (NebulaBatchOutputFormat.this) {
-                    if (!closed) {
-                        commit();
+                    if (!closed && commitException == null) {
+                        try {
+                            commit();
+                        } catch (Exception e) {
+                            commitException = e;
+                        }
```

Comment:
We need to get the exception raised from the scheduler thread. Otherwise, the exception is ignored. This approach is inspired by the official Flink JDBC connector.

Reply:
agree